src/CRUD.js

```diff
@@ -1,335 +1,301 @@
- * @param   {Object} obj - new entry
- * @returns {Object}     - object (obj)
- */
-Database.prototype.update = function ( id, obj ) {
-  if (this.data.indexOf(id) !== -1) {
-    this.destroyIndex(id); // First destroy existing index for object
-    obj[this.conf.uniqueKey] = id;
-    this.conf.driver.setItem(id, obj); // Override object
-    this.buildIndex(obj); // Rebuild index
-    return obj;
+  /**
+   * Updating an entry
+   * @param   {Number} id  - unique key of the document to update
+   * @param   {Object} obj - new entry
+   * @returns {Object}     - object (obj)
+   */
+  update( id, obj ) {
+    if (this.data.indexOf(id) !== -1) {
+      this.destroyIndex(id); // First destroy existing index for object
+      obj[this.conf.uniqueKey] = id;
+      this.conf.driver.setItem(id, obj); // Override object
+      this.buildIndex(obj); // Rebuild index
+
+      return obj;
+    }
   }
-};
 
-/**
- * Deleting an entry
- * @param  {Number|Object} arg - unique ID or object to look for before deleting matching entries
- * @returns {Boolean}           - operation status
- */
-Database.prototype.delete = function ( arg ) {
-  // If passing an object, search and destroy
-  if (Object.prototype.toString.call(arg) === '[object Object]') {
-    this.findAndDelete(arg);
-  // If passing an id, destroy id
-  } else {
-    if (this.data.indexOf(arg) !== -1) {
+  /**
+   * Deleting an entry
+   * @param  {Number|Object} arg - unique ID or object to look for before deleting matching entries
+   * @returns {Boolean}           - operation status
+   */
+  delete( arg ) {
+    // If passing an object, search and destroy
+    if (typeof arg === 'object' && arg !== null) {
+      this.findAndDelete(arg);
+    // If passing an id, destroy id
+    } else if (this.data.indexOf(arg) !== -1) {
       this.data.splice(this.data.indexOf(arg), 1);
       this.destroyIndex(arg);
       this.conf.driver.removeItem(arg);
       this.conf.driver.setItem('__data', this.data.join(','));
+
       return this.data.indexOf(arg) === -1;
     }
   }
-};
 
-/**
- * Find and delete
- * @private
- * @param   {Object}  obj - the object of properties/values to look for
- * @returns {Boolean}     - operation status
- */
-Database.prototype.findAndDelete = function ( obj ) {
-  var id, entries = this.find(obj), length = this.data.length;
-  for(var i = 0; i < entries.length; i++) {
-    id = entries[i][this.conf.uniqueKey];
-    if (this.data.indexOf(id) !== -1) {
-      this.data.splice(this.data.indexOf(id), 1);
-      this.destroyIndex(id);
-      this.conf.driver.removeItem(id);
-      this.conf.driver.setItem('__data', this.data.join(','));
+  /**
+   * Find and delete
+   * @private
+   * @param   {Object}  obj - the object of properties/values to look for
+   * @returns {Boolean}     - operation status
+   */
+  findAndDelete( obj ) {
+    const entries = this.find(obj);
+    const length = this.data.length;
+
+    for (let i = 0; i < entries.length; i++) {
```

Comment:
Any reason to not just `entries.forEach`? You even have `this` bound to your scope with an arrow function.